x-pack/plugins/lens/public/datatable_visualization/sorting.test.tsx

```diff
@@ -0,0 +1,179 @@
+          input: [now, 0, now - 150000],
+          output: [0, now - 150000, now],
+          direction,
+          type: 'date',
+        });
+      });
+    }
+  });
+
+  describe('String or anything else as string', () => {
+    for (const direction of ['asc', 'desc'] as const) {
+      it(`should provide the string criteria for terms values (${direction})`, () => {
+        testSorting({
+          input: ['a', 'b', 'c', 'd', '12'],
+          output: ['12', 'a', 'b', 'c', 'd'],
+          direction,
+          type: 'string',
+        });
+      });
+
+      it(`should provide the string criteria for other types of values (${direction})`, () => {
+        testSorting({
+          input: [true, false, false],
+          output: [false, false, true],
+          direction,
+          type: 'boolean',
+        });
+      });
+    }
+  });
+
+  describe('IP sorting', () => {
+    for (const direction of ['asc', 'desc'] as const) {
+      it(`should provide the IP criteria for IP values (IPv4 only values) - ${direction}`, () => {
+        testSorting({
+          input: ['127.0.0.1', '192.168.1.50', '200.100.100.10', '10.0.1.76', '8.8.8.8'],
+          output: ['8.8.8.8', '10.0.1.76', '127.0.0.1', '192.168.1.50', '200.100.100.10'],
+          direction,
+          type: 'ip',
+        });
+      });
+
+      it(`should provide the IP criteria for IP values (IPv6 only values) - ${direction}`, () => {
+        testSorting({
+          input: [
+            'fc00::123',
+            '::1',
+            '2001:0db8:85a3:0000:0000:8a2e:0370:7334',
+            '2001:db8:1234:0000:0000:0000:0000:0000',
+            '2001:db8:1234::', // equivalent to the above
+          ],
+          output: [
+            '::1',
+            '2001:db8:1234::',
+            '2001:db8:1234:0000:0000:0000:0000:0000',
+            '2001:0db8:85a3:0000:0000:8a2e:0370:7334',
+            'fc00::123',
+          ],
+          direction,
+          type: 'ip',
+        });
+      });
+
+      it(`should provide the IP criteria for IP values (mixed values) - ${direction}`, () => {
+        // A mix of IPv4, IPv6, IPv4 mapped to IPv6
+        testSorting({
+          input: [
+            'fc00::123',
+            '192.168.1.50',
+            '::FFFF:192.168.1.50', // equivalent to the above with the IPv6 mapping
+            '10.0.1.76',
+            '8.8.8.8',
+            '::1',
+          ],
+          output: [
+            '::1',
+            '8.8.8.8',
+            '10.0.1.76',
+            '192.168.1.50',
+            '::FFFF:192.168.1.50',
+            'fc00::123',
+          ],
+          direction,
+          type: 'ip',
+        });
+      });
+
+      it(`should provide the IP criteria for IP values (mixed values with invalid "Other" field) - ${direction}`, () => {
+        testSorting({
+          input: ['fc00::123', '192.168.1.50', '10.0.1.76', '8.8.8.8', '::1', 'Other'],
+          output: ['Other', '::1', '8.8.8.8', '10.0.1.76', '192.168.1.50', 'fc00::123'],
```

Comment:
I noticed that `Other` was sorted to the beginning of the list- I kind of expected it to be sorted in a special way, always being at the end of the list regardless of the sort order. What do you think?

Reply:
Possible, sure.
It's really up to us to decide a value out of the range I guess: the current implementation, as I saw it, mimics the "blank" values on numeric fields, which are casted to `0`.
Putting it at last should be straightforward as it knows already the direction.